### DISCUSSION TITLE
Docker fees

### DISCUSSION THREAD

**Discussion by evalieungh** [NorESMhub/noresm-land-sites-platform#142] [Announcements]
Hi everyone. @ingealthuizen recently tried to install the LSP but was told by the IT department that they would need to pay the fee because it's a work computer and the company is too large for the free option. 

See <https://www.docker.com/pricing/>

Since our software is free, open-access and educational/scientific I thought that would extend to users as well but maybe not. Other users in e.g. research institutions (like NORCE, NINA, NIBIO etc in Norway) that may want to use our platform will likely run into the same problem when they try to install Docker. Using personal computers should still be fine, but to install Docker on company-owned computers might become a problem. If the user is aiming to do a lot of work with the LSP and models it might be worth it, but if we e.g. hold a workshop for our colleagues in NINA or another company it will be a bit much to require ~650 NOK per person for just testing purposes... 

Any ideas for how to get around it? We could ask future workshop participants to use their personal laptops (if they have one), but it's not optimal.

**Comment by huitang-earth** [comment]
This is really annoying. Maybe, we should use singularity container (non profession version is free at least). Or, we should promote the use of cloud services which have docker (e.g., NREC, GALAXY). The cloud service may have paid for docker, so the users don't need to pay for that.   

Supercomputer in Finland use singularity container. I am going to test if it works for the platform.  I will update you if it works.

**Comment by ka7eh** [comment]
Looks like there was a misunderstanding by the mentioned IT department. What they're looking at is subscription to services provided by the company that maintains docker. We're not using any of those. Even the images are somewhere else (GitHub Container Registry) and not on Docker Registry.

The software part is totally separate from the service, and it's available for free and to everyone (I don't think its license is open-source, though). There's only one type of Docker Engine available to everyone, regardless of size or type of user.
So all they need to do is just install the engine, maybe with some bonus addons like the management GUI you get on Windows(?) and Mac.
Here's the documentation link for installing the software: https://docs.docker.com/desktop/.

As Hui mentioned, there are other engines, like Singularity, capable of managing images and containers.
Some extra scripts can be provided to provide support for other engines, but it will take some time to figure out the options, specially when it comes to orchestration of services, which is currently handled by docker-compose.

**Comment by evalieungh** [comment]
What a relief! In that case it seems reasonable to keep using Docker for now and then stay in touch with NCAR about what they are using. If Docker puts up paywalls for us in the future they will also likely look at other solutions.
